src/modules/vtol_att_control/standard.cpp

```diff
@@ -283,46 +289,75 @@ void Standard::update_transition_state()
+			if (elapsed > _params_standard.front_trans_time_min) {
+				_v_att_sp->pitch_body = _fw_virtual_att_sp->pitch_body;
+			}
+
+			// after ramp up start respecting FW navigation
+			if (ramp_pct >= 1.0f) {
+				_v_att_sp->roll_body = _fw_virtual_att_sp->roll_body;
+				_v_att_sp->yaw_body = _fw_virtual_att_sp->yaw_body;
+			}
+
+			// no blending if altitude control is enabled;
+			mc_weight = 1.0f;
+
+		} else {
 
-		// do blending of mc and fw controls if a blending airspeed has been provided and the minimum transition time has passed
-		if (_airspeed_trans_blend_margin > 0.0f &&
-		    _airspeed->indicated_airspeed_m_s >= _params_standard.airspeed_blend &&
-		    (float)hrt_elapsed_time(&_vtol_schedule.transition_start) > (_params_standard.front_trans_time_min * 1000000.0f)
-		   ) {
-			mc_weight = 1.0f - fabsf(_airspeed->indicated_airspeed_m_s - _params_standard.airspeed_blend) /
-				    _airspeed_trans_blend_margin;
-			// time based blending when no airspeed sensor is set
-
-		} else if (_params_standard.airspeed_disabled &&
-			   hrt_elapsed_time(&_vtol_schedule.transition_start) < (_params_standard.front_trans_time_min * 1e6f) &&
-			   hrt_elapsed_time(&_vtol_schedule.transition_start) > ((_params_standard.front_trans_time_min / 2.0f) * 1e6f)
-			  ) {
-			mc_weight = 1.0f - ((float)(hrt_elapsed_time(&_vtol_schedule.transition_start) - ((
-							    _params_standard.front_trans_time_min / 2.0f) * 1000000.0f)) /
-					    ((_params_standard.front_trans_time_min / 2.0f) * 1000000.0f));
+			// do blending of mc and fw controls if a blending airspeed has been provided and the minimum transition time has passed
+			const float airspeed_trans_blend_margin = _params_standard.airspeed_trans - _params_standard.airspeed_blend;
 
+			if ((airspeed_trans_blend_margin > 0.0f) &&
+			    (_airspeed->indicated_airspeed_m_s >= _params_standard.airspeed_blend) &&
+			    (elapsed > _params_standard.front_trans_time_min)
+			   ) {
+
+				mc_weight = 1.0f - fabsf(_airspeed->indicated_airspeed_m_s - _params_standard.airspeed_blend) /
+					    airspeed_trans_blend_margin;
+
+			} else if (_params_standard.airspeed_disabled &&
+				   (elapsed < _params_standard.front_trans_time_min) &&
+				   (elapsed > (_params_standard.front_trans_time_min / 2.0f))) {
+
+				// time based blending when no airspeed sensor is set
+				const float half_trans_time = _params_standard.front_trans_time_min / 2.0f;
+
+				mc_weight = 1.0f - (elapsed - half_trans_time) / half_trans_time;
+			}
+
+			// ramp up FW_PSP_OFF
+			_v_att_sp->pitch_body = _params_standard.pitch_setpoint_offset * (1.0f - mc_weight);
+		}
+
+		// after ramp up disable MC YAW
+		if (ramp_pct >= 1.0f) {
+			_v_att_sp->disable_mc_yaw_control = true;
 		}
 
-		// ramp up FW_PSP_OFF
-		_v_att_sp->pitch_body = _params_standard.pitch_setpoint_offset * (1.0f - mc_weight);
+		_pusher_throttle = math::constrain(trans_throttle, 0.0f, ramp_pct);
```

Comment:
If `trans_throttle` is smaller than 1.0, `pusher_throttle` will reach the `trans_throttle` before the end of the ramp. We should use `pusher_throttle = trans_throttle*ramp_pct` instead.

Reply:
Actually, I was wondering if we need transition throttle at all to be honest. The FW controller already has a maximum. VTOL could still control the smooth ramp up for now, but otherwise wouldn't have to know or care about the pusher specifics.

Reply:
It would be nice to let TECS decide of the throttle, but we should be careful about integrator windup and saturations.

Reply:
Yes I agree. It still can't be much worse than the current situation. https://github.com/PX4/Firmware/issues/8508

![image](https://user-images.githubusercontent.com/84712/35924434-0f82a032-0bf1-11e8-9e4d-0c1a54727ca1.png)